File: music-killer-java/music-killer-pojo/src/main/java/xyz/jdynb/dto/QueryPlaylistParams.java
package xyz.jdynb.dto;

import com.fasterxml.jackson.annotation.JsonAlias;
import io.swagger.v3.oas.annotations.Hidden;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 查询歌单所需的参数
 */
@EqualsAndHashCode(callSuper = true)
@Data
@Builder
@Schema(description = "查询歌单参数")
public class QueryPlaylistParams extends PageParams {

    @Schema(description = "查询id类型", example = "1,rcm")
    private String id;

    @Schema(description = "歌单id", example = "1")
    private String pid;

    @Schema(description = "排序", example = "new,hot", nullable = true)
    private String order;
}

File: music-killer-android/app/src/main/java/xyz/jdynb/music/model/QueryPlayListParams.kt
package xyz.jdynb.music.model

import kotlinx.serialization.Serializable

/**
 * 查询歌单所需的参数
 */
@Serializable
data class QueryPlayListParams(
  /**
   * 查询标识
   */
  val id: String = "",
  /**
   * 页码
   */
  val pageNo: Int = 1,
  /**
   * pageSize
   */
  val pageSize: Int = 6,
  /**
   * 排序规则
   */
  val order: String = "",
  /**
   * 歌单 id
   */
  val pid: Long = 0,
)
